### PR TITLE
Fix for ansible local provisioner error under Windows. `playbook` does not exist on the guest (#6740)

### LINKED ISSUE #6740: Windows 8.1. Ansible local provisioner error. `playbook` does not exist on the guest
`vagrant provision` output:
```
There are errors in the configuration of this machine. Please fix
the following errors and try again:

ansible local provisioner:
* `playbook` does not exist on the guest: C:/vagrant/playbook.yml
```
Working directory: `C:\Users\user\Desktop\RoR`

Environment:
 * Windows 8.1
 * Vagrant 1.8.1
 * Vagrant was installed in `C:\HashiCorp\Vagrant`
 * VirtualBox 5.0.10

Vagrantfile (`C:\Users\user\Desktop\RoR\Vagrantfile`)
```
VAGRANTFILE_API_VERSION = "2"

Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
  config.vm.box = "ubuntu/trusty64"

  config.vm.network "forwarded_port", guest: 80, host: 8080

  config.vm.provision "ansible_local" do |ansible|
    ansible.playbook = "playbook.yml"
  end
end
```

playbook.yml (`C:\Users\user\Desktop\RoR\playbook.yml`)
```
---
- hosts: all
  sudo: true
  tasks:
    - name: update apt cache
      apt: update_cache=yes
```

### PR DESCRIPTION
**Pull request by char16t**
Fix for issue #6740

### CODE REVIEW

File: plugins/provisioners/ansible/config/guest.rb [hunk 1]
```diff
@@ -37,7 +37,11 @@ def validate(machine)
         protected
 
         def check_path(machine, path, test_args, error_message_key = nil)
-          remote_path = Pathname.new(path).expand_path(@provisioning_path)
+          if (/cygwin|mswin|mingw|bccwin|wince|emx/ =~ RUBY_PLATFORM) != nil
+            remote_path = File.join(@provisioning_path, path)
+          else
+            remote_path = Pathname.new(path).expand_path(@provisioning_path)
+          end
```
Comment (gildegoma) [inline plugins/provisioners/ansible/config/guest.rb]: This patch won't work with an absolute path (e.g. `/myapp/playbook.yml`), and I am personally not sure that the `RUBY_PLATFORM` condition is a good way to go (`RUBY_PLATFORM` is only [used once](https://github.com/mitchellh/vagrant/blob/b721eb62cfbfa93895d0d4cf019436ab6b1df05d/lib/vagrant/util/subprocess.rb#L133) so far in the whole vagrant code base, and with a different purpose).

I think that we'd better use the same approach as in #542 and do the following:

```ruby
        def check_path(machine, path, test_args, error_message_key = nil)
          remote_path = File.expand_path(path, @provisioning_path)
          
          # Remove drive letter if running on a Windows host
          remote_path = remote_path.gsub(/^[a-zA-Z]:/, "")

          ...
```
Note that `Pathname.new(path).expand_path(@provisioning_path)` is replaced by `File.expand_path(path, @provisioning_path)` for performance and stability reasons (see https://github.com/mitchellh/vagrant/pull/6529#discussion_r45272058).

**Comment by gildegoma** [comment]
@char16t could you please update your PR as commented above?

**Comment by char16t** [comment]
@gildegoma Done

**Comment by char16t** [comment]
Now it looks okay?

**Comment by char16t** [comment]
@gildegoma If you are worried about the cleanliness of the commit history, you can close this PR.

**Comment by gildegoma** [comment]
@char16t can you please rebase to squash the commit history into a single commit? (Feel free to close and recreate a new PR, if you prefer).

I'll merge it later (this evening or tomorrow). Thanks!

**Comment by gildegoma** [comment]
@char16t thanks :)

**Pull request merged**